app/screens/lightning-address-screen/lightning-address-validation.ts

```diff
@@ -0,0 +1,40 @@
+export const SetAddressError = {
+  TOO_SHORT: "TOO_SHORT",
+  TOO_LONG: "TOO_LONG",
+  INVALID_CHARACTER: "INVALID_CHARACTER",
+  ADDRESS_UNAVAILABLE: "ADDRESS_UNAVAILABLE",
+  UNKNOWN_ERROR: "UNKNOWN_ERROR",
+} as const
+
+export type SetAddressError = (typeof SetAddressError)[keyof typeof SetAddressError]
+
+type ValidateLightningAddressResult =
+  | { valid: true }
+  | { valid: false; error: SetAddressError }
+
+export const validateLightningAddress = (
```

Comment:
is this validateLightningAddress or validateUsername? please according to backend regex/restrictions